BaikalDataGenerator/Source/render.cpp

```diff
@@ -0,0 +1,351 @@
+        else if (light.type == "ibl")
+        {
+            light_instance = ImageBasedLight::Create();
+
+            ImageBasedLight::Ptr ibl = std::dynamic_pointer_cast<
+                ImageBasedLight>(light_instance);
+
+            auto image_io(ImageIo::CreateImageIo());
+            // check that texture file is exist
+            auto texure_path = std::filesystem::path(light.texture);
+
+            if (!std::filesystem::exists(texure_path))
+            {
+                THROW_EX("textrue image doesn't exist on specidied path")
+            }
+
+            Texture::Ptr tex = image_io->LoadImage(light.texture);
+            ibl->SetTexture(tex);
+            ibl->SetMultiplier(light.mul);
+        }
+        else
+        {
+            THROW_EX("unsupported light type")
+        }
+
+        light_instance->SetPosition(light.pos);
+        light_instance->SetPosition(light.dir);
+        light_instance->SetPosition(light.rad);
+        m_scene->AttachLight(light_instance);
+    }
+}
+
+void Render::GenerateDataset(const std::vector<CameraInfo>& camera_states,
+                             const std::vector<LightInfo>& light_settings,
+                             const std::vector<int>& spp,
+                             const std::filesystem::path& output_dir,
+                             bool gamma_corection_enabled)
+{
+    using namespace RadeonRays;
+
+    if (!std::filesystem::is_directory(output_dir))
+    {
+        THROW_EX("incorrect output directory signature");
+    }
+
+    // check if number of samples to render wasn't specified
+    if (spp.empty())
+    {
+        return;
+    }
+
+    const int iter_number = *(std::max_element(spp.begin(), spp.end()));
+
+    SetLight(light_settings);
+
+    int cam_index = 1;
+    for (const auto &cam_state: camera_states)
+    {
+        // create camera if it wasn't  done earlier
+        if (!m_camera)
+        {
+            m_camera = Baikal::PerspectiveCamera::Create(cam_state.at,
+                                                         cam_state.pos,
+                                                         cam_state.up);
+
+            m_camera->SetSensorSize(RadeonRays::float2(0.036f, 0.036f));
+            m_camera->SetDepthRange(RadeonRays::float2(0.0f, 100000.f));
+
+            m_scene->SetCamera(m_camera);
+        }
+
+        UpdateCameraSettings(cam_state);
+
+        for (const auto& output: m_outputs)
+        {
+            output->Clear(RadeonRays::float3());
+        }
+
+        // recompile scene cause of changing camera pos and settings
+        m_controller->CompileScene(m_scene);
+        auto& scene = m_controller->GetCachedScene(m_scene);
+
+        for (auto i = 1; i <= iter_number; i++)
+        {
+            m_renderer->Render(scene);
+
+            if (std::find(spp.begin(), spp.end(), i) != spp.end())
+            {
+                for (const auto& output : outputs_collection)
+                {
+                    SaveOutput(output,
```

Comment:
Depth and Normals are the same for all spp values. So we should save only one image. For example,
cam_1_view_shading_depth.png and cam_1_view_shading_normal.png